sdk/communication/Azure.Communication.PhoneNumbers/src/SipRouting/SipRoutingClient.cs

```diff
@@ -359,15 +359,19 @@ protected SipRoutingClient()
             try
             {
                 var currentConfig = _restClient.GetSipConfiguration(cancellationToken);
+                var newTrunks = trunks.ToDictionary(x => x.Fqdn, x => x);
 
                 if (currentConfig.Value.Trunks.Count > 0)
                 {
-                    var configToRemove = new SipConfiguration(currentConfig.Value.Trunks.Keys.ToDictionary(x => x, x => (SipTrunk)null));
-                    _restClient.PatchSipConfiguration(configToRemove, cancellationToken);
+                    var trunkFqdnsToRemove = currentConfig.Value.Trunks.Keys.Where(x => !newTrunks.ContainsKey(x));
+
+                    foreach (var fqdn in trunkFqdnsToRemove)
+                    {
+                        newTrunks.Add(fqdn, null);
```

Comment:
Please correct me if I'm wrong but seems that setting `null` to the FDQN would actually remove the configuration. Maybe it's worth revisiting this approach, it seems to be a tricky behavior that's very subtle.

Reply:
@jiriburant FYI